test/e2e/emptydir_test.go

```diff
@@ -0,0 +1,83 @@
+// +build emptydir
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/spoof"
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+
+	. "knative.dev/serving/pkg/testing/v1"
+)
+
+// TestEmptyDirVolume tests empty dir volume support.
+func TestEmptyDirVolume(t *testing.T) {
+	if !test.ServingFlags.EnableAlphaFeatures {
+		t.Skip()
+	}
+
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.EmptyDir,
+	}
+
+	test.EnsureTearDown(t, clients, &names)
+
+	t.Log("Creating a new Service")
+
+	quantity := resource.MustParse("100M")
+	withVolume1 := WithVolume("data", "/data", corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{
+			Medium:    "Memory",
+			SizeLimit: &quantity,
+		},
+	})
+
+	withVolume2 := WithVolume("cache", "/cache", corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{},
```

Comment:
I dont think we actually try reading/writing to this, but I guess we just want to test the validation allows this since we tested read/write on the other volume?

Reply:
Yes to show that we can have multiple ones.